From 3cdad9183aefc422d19271a09ad01f7dda1f0cde Mon Sep 17 00:00:00 2001
From: Jan Beulich <jbeulich@suse.com>
Date: Thu, 10 Mar 2016 16:47:02 +0100
Subject: [PATCH] x86/xstate: undo bogus adjustment to xsave()

This reverts an unintended change in commit 879b44b041 ("x86/fpu: add
a per-domain field to set the width of FIP/FDP"), which I had done
intermediately while fixing the build issue: After having reverted that
adjustment I must have forgotten to "git add" the adjustment.

Signed-off-by: Jan Beulich <jbeulich@suse.com>
Reviewed-by: Andrew Cooper <andrew.cooper3@citrix.com>
diff --git a/xen/arch/x86/xstate.c b/xen/arch/x86/xstate.c
index aa6d678..8766701 100644
--- a/xen/arch/x86/xstate.c
+++ b/xen/arch/x86/xstate.c
@@ -117,7 +117,7 @@ void xsave(struct vcpu *v, uint64_t mask)
                            : "=m" (*ptr)
                            : "a" (lmask), "d" (hmask), "D" (ptr) );
 
-        if ( !(ptr->xsave_hdr.xstate_bv & XSTATE_FP) ||
+        if ( !(mask & ptr->xsave_hdr.xstate_bv & XSTATE_FP) ||
              /*
               * AMD CPUs don't save/restore FDP/FIP/FOP unless an exception
               * is pending.
